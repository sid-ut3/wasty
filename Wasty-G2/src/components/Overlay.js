/*
Overlay is a splashscreen-like component that is displayed at full width and
height in front of the application.

Typical use-cases are:

- The user is not connected to the internet.
- The user is not logged in.
- The server is not responding.

In each of these cases the application can't function, hence the need for an
overlay to tell the user why.
*/

import React, { Component } from 'react'
import { StyleSheet, View } from 'react-native'

import Icon from 'react-native-vector-icons/FontAwesome'

import AppText from './AppText'
import { colors } from '../style'

export default class Overlay extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Icon name={this.props.iconLabel} size={100} color={colors.secondary} />
        <View style={{height: 30}} />
        <AppText style={StyleSheet.flatten(styles.message)}>
          {this.props.message}
        </AppText>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  message: {
    color: colors.secondary,
    fontSize: 30,
    alignSelf: 'center'
  },
  container: {
    alignItems: 'center',
    backgroundColor: colors.primary,
    flex: 1,
    justifyContent: 'center'
  }
})

Overlay.propTypes = {
  iconLabel: React.PropTypes.string,
  message: React.PropTypes.string
}
