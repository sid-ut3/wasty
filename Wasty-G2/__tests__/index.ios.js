import 'react-native'
import React from 'react'

import renderer from 'react-test-renderer'

import Overlay from '../src/components/Overlay'

it('renders correctly', () => {
  const tree = renderer.create(
    <Overlay iconLabel='bolt' message='Vous Ãªtes hors-ligne' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
