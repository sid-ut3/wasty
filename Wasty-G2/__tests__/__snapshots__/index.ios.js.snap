exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "seagreen",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 200,
        "justifyContent": "space-between",
        "width": 200,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "darkseagreen",
            "fontSize": 100,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Nunito",
          },
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "darkseagreen",
            "fontSize": 30,
          }
        }>
        Vous êtes hors-ligne
      </Text>
    </Text>
  </View>
</View>
`;
